lua/entities/gmod_wire_expression2/core/custom/cfc_e2_lib.lua

```diff
@@ -0,0 +1,140 @@
+E2Lib.RegisterExtension("cfc_e2_lib", true)
+
+-- PvP Functions
+e2function number entity:playerIsInPvp()
+    if not IsValid(this) or not this:IsPlayer() then return 0 end
+    if this:GetNWBool("CFC_PvP_Mode", false) == true then return 1 end
+    return 0
+end
+
+e2function number entity:playerIsInBuild()
+    if not IsValid(this) or not this:IsPlayer() then return 0 end
+    if this:GetNWBool("CFC_PvP_Mode", false) == false then return 1 end
+    return 0
+end
+
+e2function string entity:getPvpStatus()
+    if not IsValid(this) or not this:IsPlayer() then return end
+    if this:GetNWBool("CFC_PvP_Mode", false) == true then return "pvp" end
+    return "build"
+end
+
+e2function table getPvpers()
+    local pvpers = {}
+    for _, v in pairs(player.GetHumans()) do
+        if v:GetNWBool("CFC_PvP_Mode", false) then pvpers[#pvpers + 1] = v end
```

Comment:
Is the `+ 1` necessary if everything is 1-indexed?

Reply:
Yes, because when the array is empty, `#pvpers` will be 0.
Here's some test code/output:
```lua
arr = {}
for i = 1, 10 do
	arr[#arr + 1] = i
end

for i = 1, 10 do
	print(i .. ": " .. arr[i])
end
```
Output:
```
1: 1
2: 2
3: 3
4: 4
5: 5
6: 6
7: 7
8: 8
9: 9
10: 10
```

Without the +1:
`[STEAM_0:1:28482516]iLikeYoBraids:8: attempt to concatenate a nil value`

Because all of the elements are being written to the 0th index.